drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c

```diff
@@ -519,11 +528,72 @@ static int wifi_nrf_drv_main_zep(const struct device *dev)
 	return -1;
 }
 
+int wifi_nrf_if_start(const struct device *dev)
+{
+	struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = dev->data;
+	struct wifi_nrf_ctx_zep *rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+
+	printk("START \n");
+
+	/* Disallow if MAC default_mac_addr is set */
+	/* If using default mac address, keep interface down */
+	if (memcmp(&rpu_ctx_zep->mac_addr, default_mac_addr,
```

Comment:
can't we just use a boolean flag instead of comparing with default MAC array? Is there an advantage to having defined default MAC as we just quit.

Reply:
We do not quit. We initialize the system with default MAC address .. 

> If the MAC address is NOT programmed in the OTP the default interface should come up with a "default" MAC address
> If  the boot-up has happened with a "default" MAC address (case 1.2 above) then regular Wi-Fi operation should not be allowed until the user has modified the MAC address.

Reply:
ok, but we still do not allow Wi-Fi operation, so, the point still holds, we can just use a flag to check if user has set MAC or not.